packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx

```diff
@@ -246,12 +246,7 @@ export const DataSourceHttpSettings: React.FC<HttpSettingsProps> = (props) => {
             </div>
           )}
 
-          {dataSourceConfig.access === 'proxy' && (
-            <HttpProxySettings
-              dataSourceConfig={dataSourceConfig}
-              onChange={(jsonData) => onSettingsChange({ jsonData })}
-            />
-          )}
+          {dataSourceConfig.access === 'proxy' && <HttpProxySettings {...props} />}
```

Comment:
Are we sure we should be spreading the props here? Pretty sure this will pass down the `props.onChange` directly rather than make use of the `onSettingsChange` function which wraps `props.onChange`. We will also be pushing things down to the `HttpProxySettings` component that is doesn't need to be aware of.